pkg/cmd/pulumi/import.go

```diff
@@ -212,6 +212,22 @@ func parseImportFile(f importFile, protectResources bool) ([]deploy.Import, impo
 		errs = multierror.Append(errs, fmt.Errorf(format, args...))
 	}
 
+	// Track duplicate resource names and their types.
+	resourceNameMap := map[tokens.QName]tokens.Type{}
+	duplicateIndexes := map[int]struct{}{}
```

Comment:
I don't think duplicateIndexes is exactly what you want here? It doesn't really matter what position things are in the resource list, we just need to worry about the set of names used.

Reply:
This is to try to preserve the original resource name that was in the JSON.

I make a first pass to identify the resources that need to have their names updated. If I don't do this, a resource with a unique name declared later on may conflict with one of the earlier renamed resources and have to be renamed leading to 2 renames instead of 1.